### PR TITLE
Remove queue configuration from conduit

### PR DESCRIPTION
**Pull request by briprowe**
We're going to be using a centralized system to manage the queue
topology.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(defproject com.farmlogs.conduit "0.1.1"
+(defproject com.farmlogs.conduit "0.2.0-SNAPSHOT"
```
Comment (mayankag) [inline project.clj]: remove snapshot before merging.

File: project.clj [hunk 2]
```diff
@@ -14,4 +14,8 @@
                     [org.slf4j/jul-to-slf4j "1.7.7"]
                     [org.slf4j/jcl-over-slf4j "1.7.7"]
                     [org.slf4j/log4j-over-slf4j "1.7.7"]
-                    [criterium "0.4.3"]]}})
+                    [criterium "0.4.3"]]}}
+  :repositories {"farmlogs-internal"
```
Comment (mayankag) [inline project.clj]: is this not a public repo?
How am I able to see these changes @kochb ?

Comment (whalesalad) [inline project.clj]: This is a public repo @mayankag! Hello from Ann Arbor! Haha.

Comment (mayankag) [inline project.clj]: thats what I thought that it is a public repo.
but then what is this change:

```:url s3p://fl-maven-repo/mvn```

isn't this super specific to farmlogs?

**Comment by micmarsh** [comment]
haha, welcome back @mayankag

**Comment by micmarsh** [comment]
Is there a write-up somewhere of the proposed new queue management system for inquiring eyes to take a gander at?

**Comment by briprowe** [comment]
@micmarsh New queue management system: https://github.com/FarmLogs/terraform-rabbitmq

File: project.clj (hunk 2 above)
Comment (briprowe) [inline project.clj]: It's there for publishing `SNAPSHOTS` for internal use.

I'll modify it to be Sonatype's OSS repo.

Comment (briprowe) [inline project.clj]: In fact after looking at this for a bit, I'd say the value of setting up Sonatype for publishing `SNAPSHOTS` is pretty low. I'm going to leave the FL internal repo here.

File: src/com/farmlogs/conduit/subscription.clj [hunk 3]
```diff
@@ -34,49 +32,45 @@
...
-  [conn prefetch-count {:keys [exchange-name exchange-type] :as config}]
-  (let [chan (rmq.chan/open conn)]
-    (rmq.basic/qos chan prefetch-count)
-    (rmq.exch/declare chan exchange-name exchange-type config)
-    (rmq.queue/declare chan (:queue-name config) config)
-    (rmq.queue/bind chan (:queue-name config) exchange-name config)
-    chan))
+  [conn prefetch-count]
+  (doto (rmq.chan/open conn)
+    (rmq.basic/qos prefetch-count)))
```
Comment (kochb) [inline src/com/farmlogs/conduit/subscription.clj]: How are we ensuring that the necessary rabbitmq structures have been created under the new architecture?

Comment (briprowe) [inline src/com/farmlogs/conduit/subscription.clj]: It's being handled by a terraform plugin: https://github.com/FarmLogs/terraform-rabbitmq

This addresses #5 here: https://farmlogs.atlassian.net/wiki/display/ENG/Workers+Failing

The worker will not start if the queue it's trying to consume from doesn't exist.

**Comment by micmarsh** [comment]
looks good +1

File: src/com/farmlogs/conduit/subscription.clj (hunk 3 above)
Comment (kochb) [inline src/com/farmlogs/conduit/subscription.clj]: I don't think we've explained this thoroughly enough yet, we need to put together better documentation - this is how to use the plugin, but which repository do I implement the transformation in?

Once it's clear how people should be creating a worker that uses a new rabbitmq resource I'll be good for a +1.

Comment (fraburnham) [inline src/com/farmlogs/conduit/subscription.clj]: In the short term manually creating queues isn't difficult. Long term we plan to have an automated solution, but I don't think not knowing what that solution will be should stop us from dealing with low hanging fruit in rmq workers.

**Comment by fraburnham** [comment]
+1

**Pull request merged**